lesson-2/web-store/src/pages/admin_page.js

```diff
@@ -0,0 +1,69 @@
+import React, {Component} from 'react';
+import {AdminItem} from './admin_item';
+import {Orders} from './../components/shared/orders_button.js';
+import {Logout} from './../components/shared/logout';
+
+
+export class Admin extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {inputValue: ''};
+        this.handleChange = this.handleChange.bind(this);
+        this.addItem = this.addItem.bind(this);
+    }
+
+    handleChange(event) {
+        this.setState({inputValue: event.target.value});
+    }  
+
+    addItem(){
+        this.props.addItem(this.state.inputValue);
+    }
+
+    render() {
+        const items = this.props.items;
+        const returnListOfItems = (items) => {
```

Comment:
you could create a method instead of re-creating function every time component re-renders